### ISSUE TITLE
Rails 5 reload deprecation message

### ISSUE THREAD

**Issue by rosskevin** [state-machines/state_machines-audit_trail#6]
It took me a while to find this, finally had to raise an error in the activerecord code to trace it.

`DEPRECATION WARNING: Passing an argument to force an association to reload is now deprecated and will be removed in Rails 5.1. Please call reload on the parent object instead.` 

```
gems/activerecord-5.0.0/lib/active_record/associations/singular_association.rb:8:in `reader'
gems/bullet-5.2.0/lib/bullet/active_record5.rb:182:in `reader'
gems/activerecord-5.0.0/lib/active_record/associations/builder/association.rb:111:in `source'
gems/state_machines-audit_trail-1.0.2/lib/state_machines/audit_trail/backend.rb:64:in `resolve_context'
gems/state_machines-audit_trail-1.0.2/lib/state_machines/audit_trail/backend.rb:22:in `block in log'
gems/state_machines-audit_trail-1.0.2/lib/state_machines/audit_trail/backend.rb:21:in `each'
gems/state_machines-audit_trail-1.0.2/lib/state_machines/audit_trail/backend.rb:21:in `log'
gems/state_machines-audit_trail-1.0.2/lib/state_machines/audit_trail/transition_auditing.rb:36:in `block in audit_trail'
gems/activesupport-5.0.0/lib/active_support/callbacks.rb:398:in `instance_exec'
gems/activesupport-5.0.0/lib/active_support/callbacks.rb:398:in `block in make_lambda'
gems/activesupport-5.0.0/lib/active_support/callbacks.rb:207:in `block in halting_and_conditional'
gems/activesupport-5.0.0/lib/active_support/callbacks.rb:456:in `block in call'
gems/activesupport-5.0.0/lib/active_support/callbacks.rb:456:in `each'
gems/activesupport-5.0.0/lib/active_support/callbacks.rb:456:in `call'
gems/activesupport-5.0.0/lib/active_support/callbacks.rb:101:in `__run_callbacks__'
gems/activesupport-5.0.0/lib/active_support/callbacks.rb:750:in `_run_initialize_callbacks'
gems/activerecord-5.0.0/lib/active_record/core.rb:322:in `initialize'
gems/state_machines-activerecord-0.4.0/lib/state_machines/integrations/active_record.rb:454:in `initialize'
gems/activerecord-5.0.0/lib/active_record/inheritance.rb:65:in `new'
gems/activerecord-5.0.0/lib/active_record/inheritance.rb:65:in `new'
gems/activerecord-5.0.0/lib/active_record/persistence.rb:50:in `create!'
```

**Comment by rosskevin** [comment]
@seuros do you have a preference on versioning a rails 5 compatible release as well as creating any maintenance branches?  This gem is currently at 1.0.2.  

I'm working on a branch to fix this issue, but I'm thinking:
- [ ] create 1-0-stable branch (for rails 4 support)
- [ ] merge this fix and release as 2.0.0

Thoughts?

**Comment by rosskevin** [comment]
This is currently fixed (worked around really) on the `rails-5` branch.

**Comment by rosskevin** [comment]
rails-5 branch re-established with this change on the new master.

**Comment by seuros** [comment]
@rosskevin If we cant support both version with the same code. We should release version 2.x 
Version 1.x will receive fixes until rails 4.2 stop being supported.

**Comment by rosskevin** [comment]
My code change is compatible (and actually makes sense even on rails 4).

The incompatibilities come externally it seems:
1. rails 5 requires ruby 2.2.3(?)+ which we can be silent on
2. mongodb definitely has changes for rails 5

I think it is cleaner/easier to just make 1.x rails-4 compatible, 2.x updated dependencies/rails-5

The holdup on 2.x/rails 5 could be mongodb, but I don't use it so I'm not interested in maintaining that part.  I don't mind releasing without it, leaving an open issue for a contributor to pick up.

What do you think?
